src/components/AAPModal/__tests__/AnsibleAutomationPlatformModal.spec.tsx

```diff
@@ -0,0 +1,156 @@
+  metadata:
+    creationTimestamp: "2025-01-04T16:51:17Z"
+    name: sandbox-aap
+    uid: 60d90446-9c1f-4875-b00c-ac8954a67397
+  spec:
+  status:
+    URL: https://sandbox-aap-test.com
+    adminPasswordSecret: sandbox-aap-admin-password
+    adminUser: admin
+    conditions:
+    - lastTransitionTime: "2025-01-04T17:10:07Z"
+      message: ""
+      reason: ""
+      status: "False"
+      type: Failure
+    - lastTransitionTime: "2025-01-04T17:16:50Z"
+      message: Last reconciliation succeeded
+      reason: Successful
+      status: "True"
+      type: Successful
+    - lastTransitionTime: "2025-01-04T16:51:17Z"
+      message: Running reconciliation
+      reason: Running
+      status: "True"
+        `);
+
+const encode = (str: string): string => Buffer.from(str, 'binary').toString('base64');
+
+jest.mock('../../../hooks/useRegistrationService', () => {
+  const mock = {
+    getSignupData: jest.fn(),
+  };
+  return {
+    __esModule: true,
+    default: () => mock,
+  };
+});
+
+jest.mock('../../../hooks/useKubeApi', () => {
+  const mock = {
+    getAAPData: jest.fn(),
+    getSecret: jest.fn(),
+  };
+  return {
+    __esModule: true,
+    default: () => mock,
+  };
+});
+
+describe('AnsibleAutomationPlatformModal', () => {
+  let getAAPDataMock: jest.Mock;
+  let getSecretMock: jest.Mock;
+  let getSignupDataMock: jest.Mock;
+  const mockCallBack = jest.fn();
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+    jest.resetAllMocks();
+    const kubeService = useKubeApi();
+    getAAPDataMock = kubeService.getAAPData as jest.Mock;
+    getSecretMock = kubeService.getSecret as jest.Mock;
+    const signupService = useRegistrationService();
+    getSignupDataMock = signupService.getSignupData as jest.Mock;
+  });
+
+  function requiredModalTextWhenProvisioning() {
+    // title should be there
+    const modalTitle = screen.getByText('Provisioning Ansible Automation Platform (AAP) instance');
+    expect(modalTitle).toBeDefined();
+    // modal content should be there
+    const { getByText } = within(screen.getByTestId('modal-content'));
+    expect(
+      getByText(
+        'Your AAP instance might take up to 30 minutes to provision. Once ready, your instance will remain active for 12 hours.',
+      ),
+    ).toBeInTheDocument();
+    expect(screen.getByRole('link', { name: 'AAP documentation' })).toHaveAttribute(
+      'href',
+      'https://docs.redhat.com/en/documentation/red_hat_ansible_automation_platform',
+    );
+    // spinner should be there
+    const progressBar = screen.getByRole('progressbar');
+    expect(progressBar).toBeDefined();
+    // close modal button should be available
+    const button = screen.queryByText('Close');
+    button?.click();
+    expect(mockCallBack).toHaveBeenCalled();
+  }
+
+  it('modal should say that provisioning is in progress when initialStatus is unknown', () => {
+    render(<AnsibleAutomationPlatformModal onClose={mockCallBack} initialStatus="unknown" />);
```

Comment:
Something for the future. For components, it is better to use for example Cypress component testing: https://docs.cypress.io/app/component-testing/get-started

The problem with Jest and react testing library is the fact that it does not run in a "real" browser and the interaction does not match the actual user interaction happening in browser. This leads to unreliable tests that might work in the JSDOM env and not in browser.

Reply:
Surprisingly they are not that much slower. There is slightly more "upfront work" but the tests themselves are pretty rapid. I'd say on par with jest.

But yeah, not something that should be included in this PR.

Reply:
ok got it! Thanks, I'll maybe run an experiment and try to convert one of those and see how it works, just for learning/investigation purposes.